1091. Shortest Path in Binary Matrix/solution.py

```diff
@@ -1,47 +1,79 @@
 class Solution(object):
-	# Given an n x n binary matrix grid, we return the shortest clear path in 
-	# the matrix, or -1 for no clear path.
+# Given an n x n binary matrix grid, we return the shortest clear path in 
+# the matrix, or -1 for no clear path.
 
-	# A clear path is a path from the top-left cell to the bottom right cell
-	# such that:
+# A clear path is a path from the top-left cell to the bottom right cell
+# such that:
 
-	 	# 1. all visited cells are 0
-	 	# 2. all adjacent cells are different and share an edge or corner
+	# 1. all visited cells are 0
+	# 2. all adjacent cells are different and share an edge or corner
 
+	def findAdjacent(grid: List[List[int]], start: Tuple[int, int]) -> List[Tuple[int, int]]:
+	# Create method to find surrounding cells that are clear path, given a  
+	# matrix and a current location.
 
-	def findAdjacent(grid: List[List[int]], start_row: Tuple[int, int], start_col: Tuple[int, int]) -> List[Tuple[int, int]]:
-		# Create method to find surrounding cells that are clear path, given a  
-		# matrix and a current location.
+		# Define directions as tuples (-1, 0), (1, 0), (0, -1), (0, 1), 
+		# (-1, -1), (-1, 1), (1, -1), (1, 1), (-1, -1).
+		directions = [(-1, 0), (1, 0), (0, -1), (0, 1), (-1, 1), (1, -1), (1, 1), (-1,-1)]
+		clearPaths = []
 
-			# Define directions as tuples (-1, 0), (1, 0), (0, -1), (0, 1), 
-			# (-1, -1), (-1, 1), (1, -1), (1, 1).
+		# For the given location, check if there are any 0's in the 8  
+		# surrounding spots. Also check if any of these are out of bounds.
+		for i in range(len(directions)):
+			new_x = start[0] + directions[i][0]
+			new_y = start[1] + directions[i][1]
 
-			# For the given location, check if there are any 0's in the 8  
-			# surrounding spots. Also check if any of these are out of bounds.
+			if 0 <= new_x < len(grid) and 0 <= new_y < len(grid[0]) and grid[new_x][new_y] == 0 :
+				clearPaths.append(directions[i])
 
-			# Return a list of the surrounding 0's
+		# Return a list of the surrounding 0's
+		return clearPaths
 
 
 	def shortestPathBinaryMatrix(self, grid: List[List[int]]) -> int:
 		# Initalize queue for BFS to keep track of which cells are waiting to be 
 		# processed as queue = [(0,0)]. Tuples represent the difference between the
 		# matrix location and the upper leftmost cell.
+		queue = [[0,0]]
+
 		# Initialize step counter variable
+		step = 1
+
+		# Get the width and height of the matrix
+		gridWidth, gridHeight = len(grid), len(grid[0])
+
+		# Check if we're starting with 0, if not, return -1
+		if(grid[0][0] != 0):
```

Comment:
Shouldn't you also check if we're ending with a 1?

Reply:
Oh! That's a good idea. I wonder what kind of performance gain I will see from checking that, I'll test it out.

Reply:
I tried a couple of different submits using both with a check for ending with 1 and without but couldn't find anything conclusive. >_> It seems like this might increase memory usage a little bit (from 16.7 or 16.6 to 16.9 MB), but it also looks like there's a little bit of variability in memory usage between each run even if the code is the same(?).